src/userTourManager.ts

```diff
@@ -0,0 +1,102 @@
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+import { PromiseDelegate } from '@lumino/coreutils';
+
+import {
+  IUserTourManager,
+  ITour,
+  ITourManager,
+  USER_PLUGIN_ID
+} from './tokens';
+
+/**
+ * The UserTourManager is needed to manage syncing of user settings with the TourManager
+ */
+export class UserTourManager implements IUserTourManager {
+  constructor(options: IUserTourManager.IOptions) {
+    this._tourManager = options.tourManager;
+    options
+      .getSettings()
+      .then(userTours => {
+        this._userTours = userTours;
+        this._userTours.changed.connect(this._userToursChanged, this);
+        this._userToursChanged();
+        this._ready.resolve();
+      })
+      .catch(console.warn);
```

Comment:
Should we reject the ready promise in case of error?

```suggestion
      .catch(reason => {
        console.warn(reason);
        this._ready.reject();
      });
```